### DISCUSSION TITLE
Is it possible to pass NULL as a parameter value with server-side binding?

### DISCUSSION THREAD

**Discussion by kdwyer** [psycopg/psycopg#582] [Q&A] [answered]
According to the [docs](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#server-side-binding)

> Server-side binding works for normal SELECT and data manipulation statements

However attempting to pass `NULL` as a parameter value for a `WHERE ... IS NULL` clause result in an error characteristic of server-side binding not working.

```python
Python 3.11.3 (main, May 29 2023, 09:16:30) [GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import psycopg
>>> psycopg.__version__
'3.1.9'
>>> conn = psycopg.connect(dbname='test')
>>> cur = conn.cursor()
>>> cur.execute("""SELECT * FROM users WHERE encrypted IS %s""", (None,))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kev/virtual-envs/so311/lib/python3.11/site-packages/psycopg/cursor.py", line 723, in execute
    raise ex.with_traceback(None)
psycopg.errors.SyntaxError: syntax error at or near "$1"
LINE 1: SELECT * FROM users WHERE encrypted IS $1
                                               ^
```

Client-side binding, either through `ClientCursor` or statement construction using `psycopg.sql` works without error.

I couldn't find anything in the sections on types adaptation about `None`/`NULL`. Is this an omission in the docs or am I missing something?

**Comment by dvarrazzo** [comment, marked as answer]
>
> This problem has been reported by a few people. I think that IS is not a
binary operator, but IS NULL is a unary operatore. You can't write 10 IS 20.

You can use `WHERE encrypted IS NOT DISTINCT FROM %s`: DISTINCT is a proper
binary operator and you can pass it a NULL, which would have the same
semantics of IS NULL.

I will add a note in the docs.

**Reply by dvarrazzo** [reply]
Added [docs section](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#you-cannot-use-is-s).
